modules/samples/virtual_firealarm/src/org.wso2.carbon.device.mgt.iot.sample.virtual.firealarm.service.impl/src/main/java/org/wso2/carbon/device/mgt/iot/sample/virtual/firealarm/service/impl/VirtualFireAlarmControllerService.java

```diff
@@ -78,9 +85,53 @@
 	public static final String HTTP_PROTOCOL = "HTTP";
 	public static final String MQTT_PROTOCOL = "MQTT";
 
+	private static ConcurrentHashMap<String, String> deviceToIpMap = new ConcurrentHashMap<String, String>();
+	private static XMPPClient xmppClient;
+	private static MQTTClient mqttClient;
+	private static final String mqttServerSubscribeTopic = "wso2/iot/+/" + VirtualFireAlarmConstants.DEVICE_TYPE + "/+/reply";
+	private static final String iotServerSubscriber = "IoT-Server";
+
+	static{
+		String xmppServer = XmppConfig.getInstance().getXmppControlQueue().getServerURL();
+		int indexOfChar = xmppServer.lastIndexOf('/');
```

Comment:
method lastIndexOf in String class accepts int not a String. Specifying char value to place where int expects is also valid in java.
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#lastIndexOf(int)